docs/user/using-an-OIDC-provider.md

```diff
@@ -59,39 +61,77 @@ In the case of Google we can use an OAuth 2.0 client ID. You can find more infor
+The main difference in the authentication is that instead of accessing the Kubeapps service, we will be accessing an oauth2 proxy service that is in charge of authenticating users with the identity provider and injecting the required credentials in the requests to Kubeapps. There are a number of available solutions for this use-case, like [keycloak-gatekeeper](https://github.com/keycloak/keycloak-gatekeeper) and [oauth2_proxy](https://github.com/pusher/oauth2_proxy). For this guide we will use `oauth2_proxy` since it supports both OIDC and plain OAuth2 for many providers.
 
-Once the proxy is accessible, you will be redirected to your IdP. There, you will be prompted to introduce your credentials:
-
-![Proxy Login](../img/auth-proxy-login.png)
-
-After successfully logging in with your IdP, you will be redirected to Kubeapps and be authenticated with your user's OIDC token.
+Once the proxy is accessible, you will be redirected to the identity provider to authenticate. After successfully authenticating, you will be redirected to Kubeapps and be authenticated with your user's OIDC token.
 
 The next sections explain how you can deploy this proxy either using the Kubeapps chart or manually.
 
 ### Using the chart
 
-Kubeapps chart allows you to automatically deploy the proxy for you as a sidecar container if you specify the necessary flags. In a nutshell you need to enable the feature and set the client ID, secret and the IdP URL. This is an example for using Google as IdP, modify the flags below to adapt them:
+Kubeapps chart allows you to automatically deploy the proxy for you as a sidecar container if you specify the necessary flags. In a nutshell you need to enable the feature and set the client ID, secret and the IdP URL. The following examples use Google as the Identity Provider, modify the flags below to adapt them:
+
+**Example 1: Using Google OpenIDConnect**
+
+Note that the issuer url is passed as an additional flag here, together with an option to enable the cookie being set over an insecure connection for local development only:
+
+```
+helm install bitnami/kubeapps \
+  --namespace kubeapps --name kubeapps \
+  --set authProxy.enabled=true \
+  --set authProxy.provider=oidc \
+  --set authProxy.clientID=my-client-id.apps.googleusercontent.com \
+  --set authProxy.clientSecret=my-client-secret \
+  --set authProxy.cookieSecret=$(echo "not-good-secret" | base64) \
+  --set authProxy.additionalFlags="{-cookie-secure=false,-oidc-issuer-url=https://accounts.google.com}" \
+```
+
+**Example 2: Using Google OAuth2**
```

Comment:
I find this example confusing, I mean, why would I use Google Oauth2 in the general sense? I know that for GKE it makes sense to do it like that but just in that case isn't it? 

I'd have just the generic OIDC example + the new GKE one using oauth2.

Reply:
> I find this example confusing, I mean, why would I use Google Oauth2 in the general sense? I know that for GKE it makes sense to do it like that but just in that case isn't it?

Yes it's confusing, but no it makes sense to use the 'google' provider generally, it's the heading I've given that is incorrect - it's really using `oauth2-proxy`'s custom Google OIDC provider. The benefit of using the `google` provider is that it provides a richer UI experience to users (as I tried to explain below in the GKE example). One aspect of that only affects you when you're requesting extra scopes (which generally would only be on GKE)[1], another is the default button shown by `oauth2-proxy` ("Login with Google" rather than "Login with OpenID Connect") - but that is configurable too.

A few of the `oauth2-proxy` supported providers are actually using OpenID Connect and setting the IDToken in the session state. These are: gitlab, google and logingov.

Let me re-word these examples to see if I can make it much clearer.

Thanks!

[1] This might change in the future - I don't see why Google wouldn't want the same UI experience for generic OIDC logins, or it could even be a quirk of `oauth2_proxy`'s generic oidc implementation - not sure.

Reply:
Please take another look :)

Reply:
> "Login with Google"

I thought that we were redirecting by default to the IdP?